Library table: allow to ignore Track double-click

ronso0 opened the pull request:
Many times a accidental touchpad double-click has caused a PITA in live sessions because it cleared an already prepared deck.
Setting track double-click to "AutoDJ add ..." would protect the decks but pollute the AutoDJ queue.

So I'm adding the valid 'Ignore' option here. Default is untouched (load to deck).
![image](https://user-images.githubusercontent.com/5934199/96598895-a0d92080-12ef-11eb-8199-e880512e9f8c.png)

Also the options section is renamed: `Track Load Action` >> `Track Double Click Action` (as the labels extracomment="" already says)

daschuer reviewed (approved):
LGTM thank you. Waiting for CI.

daschuer commented:
@Holzhaus: unfortunately the pre-commit hock on GitHub fails. Can we revive the Travis one
and keep both of them until GitHub is stable?

I also cannot get used to the cluttered output of the GitHub version. The original text only output is much more informative at once.
Can we implement this at GitHub as well and bypass all the fancy GUI menüs?

Holzhaus commented:
> @Holzhaus: unfortunately the pre-commit hock on GitHub fails. Can we revive the Travis one
> and keep both of them until GitHub is stable?

The failure was caused by an in-progress apt mirror sync while installing clang-format. That would have happened on Travis as well. I restarted the jobs and now it works:



> I also cannot get used to the cluttered output of the GitHub version. The original text only output is much more informative at once.
> Can we implement this at GitHub as well and bypass all the fancy GUI menüs?

Hmm, I

daschuer reviewed (changes requested):
Unfortunatly MSVC fails with

```
.\src\widget\wtracktableview.cpp(343): error C2589: 'constant': illegal token on right side of '::'
..\src\widget\wtracktableview.cpp(343): error C2062: type 'unknown-type' unexpected
..\src\widget\wtracktableview.cpp(343): error C2143: syntax error: missing ';' before '{'
```

daschuer commented on src/widget/wtracktableview.cpp:
```diff
@@ -340,6 +340,10 @@ void WTrackTableView::slotMouseDoubleClicked(const QModelIndex& index) {
             static_cast<DlgPrefLibrary::TrackDoubleClickAction>(
                     doubleClickActionConfigValue);
 
+    if (doubleClickAction == DlgPrefLibrary::IGNORE) {
```
I am afraid Windows has already defined IGNORE. 
That can be fixed by using a enum class, just rename the enum value.

ronso0 replied:
I'll rename to DO_NOTHING

ronso0 committed:
Library table: allow to ignore Track double-click

ronso0 commented:
set to Draft because I overlooked that this change is also suppressing [Library],GoToItem in the tracks table.

ronso0 committed:
librarycontrol.cpp: fix slotGotToItem in tracks table

Appearantly m_pLibraryWidget->hasFocus() always returns false.
Instead check m_pLibraryWidget->getActiveView()->hasFocus()

ronso0 commented:
> set to Draft because I overlooked that this change is also suppressing [Library],GoToItem in the tracks table.

wrong.
It was just that `GoToItem` wasn't working earlier.
I included the fix from #3196 here.

@daschuer Please check again.

Holzhaus commented on src/preferences/dialog/dlgpreflibrary.h:
```diff
@@ -22,7 +22,8 @@ class DlgPrefLibrary : public DlgPreferencePage, public Ui::DlgPrefLibraryDlg  {
     enum TrackDoubleClickAction {
         LOAD_TO_DECK,
         ADD_TO_AUTODJ_BOTTOM,
-        ADD_TO_AUTODJ_TOP
+        ADD_TO_AUTODJ_TOP,
+        DO_NOTHING
     };
```
Can you make this an enum class with CamelCase members?

Also, I think `DO_NOTHING` should be renamed to `IGNORE` for consistency with the radio button.

ronso0 replied:
I can try it.
camelCase or PascalCase?

re: ignore see [@daschuer 's remark above](https://github.com/mixxxdj/mixxx/pull/3194#discussion_r509695739), the Windows CI failed earlier because of that.

Holzhaus replied:
PascalCase, sorry.

If you use an enum class, there shouldn't be a naming conflict because enum classes as properly scoped (unlink oldschool enums) and the name will be different: `TrackDoubleClickAction::Ignore` vs. `IGNORE`.

ronso0 replied:
Done.

Holzhaus replied:
I always wonder how you're understanding my review comments considering all the typos introduced by my phone's "aurocorrect" xD

ronso0 replied:
I have a built-in autocorrect filter ;|

ronso0 replied:
thinking about it... a buddy of mine has dyslexia, or heavy difficulties with writing,. I think he gave trying to do it correctly. so he's trained my phantasy and ability to abstract.. a lot!

ronso0 committed:
DlgPrefLibrary: use enum class for TrackDoubleClickAction

daschuer reviewed (approved):
LGTM, Thank you.
